server/src/main/java/org/elasticsearch/cluster/LocalNodeMasterListener.java

```diff
@@ -37,16 +39,20 @@
     /**
      * The name of the executor that the implementation of the callbacks of this lister should be executed on. The thread
      * that is responsible for managing instances of this lister is the same thread handling the cluster state events. If
-     * the work done is the callbacks above is inexpensive, this value may be
+     * the work done is the callbacks above is inexpensive, this value should be
      * {@link org.elasticsearch.threadpool.ThreadPool.Names#SAME SAME} (indicating that the callbacks will run on the same thread
      * as the cluster state events are fired with). On the other hand, if the logic in the callbacks are heavier and take
      * longer to process (or perhaps involve blocking due to IO operations), prefer to execute them on a separate more appropriate
-     * executor (eg. {@link org.elasticsearch.threadpool.ThreadPool.Names#GENERIC GENERIC}
-     * or {@link org.elasticsearch.threadpool.ThreadPool.Names#MANAGEMENT MANAGEMENT}).
+     * executor; for example management operations would use {@link org.elasticsearch.threadpool.ThreadPool.Names#MANAGEMENT MANAGEMENT}).
+     *
+     * <em>Note</em>: when using an executor with more than one thread the callbacks <i>could</i> run concurrently, so care
+     * should be taken to ensure proper locking is used for correctness.
      *
      * @return The name of the executor that will run the callbacks of this listener.
      */
-    String executorName();
+    default String executorName() {
```

Comment:
This looks never to be overridden, can we inline it (and simplify `threadPool.executor(SAME).execute(r)` to `r.run()` in the `ClusterApplierService`?

Reply:
Addressed in https://github.com/elastic/elasticsearch/pull/59932/commits/63be6ab56326f0a496a33588730130aa227bb45f